front/src/Phaser/Game/MapEditor/Tools/AreaEditorTool.ts

```diff
@@ -42,20 +44,93 @@ export class AreaEditorTool extends MapEditorTool {
     }
 
     public subscribeToRoomConnection(connection: RoomConnection): void {
-        connection.editMapMessageStream.subscribe((message) => {
-            switch (message.message?.$case) {
+        connection.editMapCommandMessageStream.subscribe((editMapCommandMessage) => {
+            const pendingCommands = this.mapEditorModeManager.getPendingCommands();
+
+            if (pendingCommands.length > 0) {
+                if (pendingCommands[pendingCommands.length - 1].id === editMapCommandMessage.id) {
+                    console.log("OUR NEWEST COMMAND CAME BACK FROM THE BACK, ACKNOWLEDGED!");
+                    pendingCommands.pop();
+                    return;
+                } else if (pendingCommands.map((command) => command.id).includes(editMapCommandMessage.id)) {
+                    console.log(
+                        "INCOMING COMMAND IS NOT THE NEWEST COMMAND WAITING FOR ACKNOWLEDGEMENT. REVERTING PENDING COMMANDS"
+                    );
+                    while (pendingCommands.length > 0) {
+                        pendingCommands.pop()?.undo();
+                    }
+                    console.log("PENDING COMMANDS CLEARED. APPLY NEWEST COMMAND FROM THE SERVER");
+                }
```

Comment:
Here, the condition in the "else if" is probably unnecessary.

You are trying to detect if a command is part of the commands that we sent. But if someone else is sending us a command, this command will not be part of the list. Yet, we still need to undo our commands to apply the command from the other user (our commands will come back later anyway)